server/schema/types/root_query_type.js

```diff
@@ -36,118 +36,127 @@ const FilterInputType = require('./filter_input_type')
+            if (validUser.loggedIn) {
+                const userId = validUser.id;
+                return User.findById(userId)
+            } else {
+                throw new Error("No one is logged in!")
             }
+        }
+    },
+    fetchShopCoffees: {
+        type: new GraphQLList(CoffeeType),
+        args: {
+          coffeeShopId: { type: GraphQLID },
+          filter: { type: FilterInputType }
         },
-        coffees: {
-            type: new GraphQLList(CoffeeType),
-            args: { filter: { type: FilterInputType } },
-            resolve(_, {filter}) {
+        resolve(_, { coffeeShopId, filter }) {
 
-                function buildFilters({ processing, roasting, flavor, price }) {
-                    
-                    const filter = (processing || roasting || flavor || price ) ? {} : null;
-                    
-                    if (processing) {
-                        filter.processing = `${processing}`;
-                    }
-                    if (roasting) {
-                        filter.roasting = `${roasting}`;
-                    }
-                    if (flavor) {
-                        filter.flavor = {$in: flavor};
-                    }
-                    if (price) {
-                        filter.price = {$gt: price[0], $lt: price[1]};
-                    }
-                    let filters = filter ? [filter] : [];
-                    return filters;
-                }
+            function buildFilters({ processing, roasting, flavor, price }) {
+              let filters = {}
 
-                let query = filter ? {$and: buildFilters(filter)} : {};
-                return Coffee.find(query)
-            }
-        },
-        coffee: {
-            type: CoffeeType,
-            args: { id: { type: new GraphQLNonNull(GraphQLID) } },
-            resolve(_, args) {
-                return Coffee.findById(args.id)
-            }
-        },
-        coffeeShops: {
-            type: new GraphQLList(require("./coffee_shop_type").CoffeeShopType),
-            resolve() {
-                return CoffeeShop.find({});
-            }
-        },
-        coffeeShop: {
-            type: require("./coffee_shop_type").CoffeeShopType,
-            args: { id: { type: new GraphQLNonNull(GraphQLID) }},
-            resolve(_, args) {
-                return CoffeeShop.findById(args.id);
+              if (processing) {
+                filters.processing = processing
+              }
+              if (roasting) {
+                filters.roasting = roasting
+              }
+              if (flavor && flavor.length !== 0) {
+                filters.flavor = { $in: flavor };
+              }
+              if (price && price.length !== 0) {
+                filters.price = { $gt: price[0], $lt: price[1] };
+              }  
+        
+              let updatedFilter = Object.keys(filters).length === 0 ? [] : [filters];
+              return updatedFilter;
             }
-        },
```

Comment:
Is this deleting search functionality? Might just be my misunderstanding of git since this removal takes place after the changes to search above.

Reply:
No, it is up in there, see right side line 80-104. When I merge, the lines you wrote the function were overlapped with the code I wrote, so I readjust the order.